common/djangoapps/util/course.py

```diff
@@ -0,0 +1,49 @@
+"""
+Utility methods related to course
+"""
+import logging
+from django.conf import settings
+
+from opaque_keys.edx.keys import CourseKey
+import re
+
+log = logging.getLogger(__name__)
+
+
+def get_lms_link_for_about_page(request, course_key):
+    """
+    Returns the url to the course about page from the location tuple.
+    """
+
+    assert isinstance(course_key, CourseKey)
+
+    if settings.FEATURES.get('ENABLE_MKTG_SITE', False):
+        if not hasattr(settings, 'MKTG_URLS'):
+            log.exception("ENABLE_MKTG_SITE is True, but MKTG_URLS is not defined.")
+            return None
+
+        marketing_urls = settings.MKTG_URLS
+
+        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
+        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
+        about_base = marketing_urls.get('ROOT', None)
+
+        if about_base is None:
+            log.exception('There is no ROOT defined in MKTG_URLS')
+            return None
+
+        # Strip off https:// (or http://) to be consistent with the formatting of LMS_BASE.
+        about_base = re.sub(r"^https?://", "", about_base)
```

Comment:
I think i found the reason why regex was used instead or urlparser. There is a test that verifies marketing site url for https edge and the ROOT marketing URL contains http as part of its domain. The url parser takes the whole domain (with https) as scheme and omits the value we needed.
For reference, (the test appears as deleted in this PR but thats because its location is changed from cms repo to common)
https://github.com/edx/edx-platform/pull/12818/files#diff-210a85f233621ab76f3b42bbce5e9c36L49